### DISCUSSION TITLE
Logging for command line app

### DISCUSSION THREAD

**Discussion by Puderzo** [j-kaltes/Juggluco#51] [Q&A]
I'm trying to troubleshoot the command line app running on a Google Cloud VM (the same one I'm running Nightscout on), which seems to work intermittently on an off. I can see the clients connect in the firewall log, but they don't seem to update streams (polls.dat timestamp does not change, and exported streams (./juggluco -B) do not show updated info) and the mirror status on the clients don't update with connection timestamps. Also, the receiver mirror does not get updated streams. I'm unsure if I've set up the mirror listeners correctly, or if there's some other issue, and logging would probably help this troubleshooting a lot. Is there a way to set up logging of incoming connections on this?

**Comment by j-kaltes** [comment]
The debug version does log, but this will contain a lot of information.
You can make it as described in the README:
```
cmake -DDEBUG=ON $srcdir
make juggluco
```
You can better look at the output of
`juggluco -l`
and in left middle menu->Mirror to the specific connection and port numbers and whether all ports are open in your firewall.
Also netstat etc give information.
By setting _Detect_ instead of the IP, you can see whether contact was made or not.
Without -i the command line version does automatically test on IP.

You can look at the data with a hex editor:
https://www.juggluco.nl/Juggluco/cmdline/data/index.html
